lib/rspec/support/spec/library_wide_checks.rb

```diff
@@ -47,11 +47,11 @@ def load_all_files(files, preamble, postamble=nil)
     files = all_lib_files - lib_test_env_files
     preamble  = ['orig_loaded_features = $".dup', preamble_for_lib]
     postamble = [
-      'loaded_features = ($" - orig_loaded_features).join("\n")',
-      "File.open('#{loaded_features_outfile}', 'w') { |f| f.write(loaded_features) }"
+      'puts ($" - orig_loaded_features).join("\n")'
     ]
 
-    load_all_files(files, preamble, postamble)
+    @loaded_features << load_all_files(files, preamble, postamble)[0]
+    load_all_files(files, [preamble_for_lib])
```

Comment:
One of my goals in setting this up was to not load all lib files twice, which is what you are doing here. It's slow so it's preferrable to do it once and leverage it for both warnings and loaded features.  Here you're loading all lib files twice instead.  I think we can get by with doing it only once with:

``` ruby
    @loaded_features, *rest = load_all_files(files, preamble, postamble)
    return "", *rest
```

Basically, load it once and capture `@loaded_features` from stdout, and return `""` for stdout in its place to satisfy the expectation.  For the warnings detection, the warnings are on stderr anyway, and if there was a weird one that showed up on stdout our loaded feature check would catch it (since it requires a whitelisted regex for each line) so there's not really risk of making this a false positive by returning `""` for stdout.  Really, we probably don't even need to assert on stdout for the warnings specs....it just seemed simple enough to do it when I set it up.